app/controllers/admin/categories_controller.rb

```diff
@@ -0,0 +1,13 @@
+module Admin
+  class CategoriesController < ApplicationController
+
+    def index
+      @categories = Category.all.where(parent_category_id: nil)
+    end
+
+    def products
+      category = Category.find_by(id: params[:category_id])
```

Comment:
what if category is not found?

Reply:
if not found redirected it to the categories page